server/gorm/deletions.go

```diff
@@ -97,8 +97,18 @@ func (c *Client) DeleteChildrenOfVersion(ctx context.Context, version *models.Ve
 		q = q.Require("VersionID", version.VersionID)
 		err := c.DeleteAllMatches(ctx, q)
 		if err != nil {
-			//return err
+			return err
 		}
 	}
 	return nil
 }
+
+// DeleteChildrenOfSpec deletes all the children of a spec.
+func (c *Client) DeleteChildrenOfSpec(ctx context.Context, spec *models.Spec) error {
+	q := c.NewQuery(models.BlobEntityName)
```

Comment:
This should probably also delete artifacts associated with the spec. I'll create an issue that we can address later (same for APIs and versions).

Reply:
👍